token/external_provider.go

```diff
@@ -0,0 +1,194 @@
+package token
```

Comment:
Let's move this to token/provider package. It helps as too avoid cycle dependencies. See https://github.com/fabric8-services/fabric8-auth/pull/136

Reply:
Done